lib/tpm2_identity_util.c

```diff
@@ -423,3 +423,43 @@ void tpm2_identity_util_calculate_outer_integrity(TPMI_ALG_HASH parent_name_alg,
             encrypted_duplicate_sensitive->size, pubname->name, pubname->size,
             protection_hmac_key->buffer, outer_hmac);
 }
+
+bool tpm2_identity_create_name(TPM2B_PUBLIC *public, TPM2B_NAME *pubname) {
+
+    /*
+     * A TPM2B_NAME is the name of the algorithm, followed by the hash.
+     * Calculate the name by:
+     * 1. Marshaling the name algorithm
+     * 2. Marshaling the TPMT_PUBLIC past the name algorithm from step 1.
+     * 3. Hash the TPMT_PUBLIC portion in marshaled data.
+     */
+
+    TPMI_ALG_HASH name_alg = public->publicArea.nameAlg;
+
+    // Step 1 - set beginning of name to hash alg
+    size_t hash_offset = 0;
+    Tss2_MU_UINT16_Marshal(name_alg, pubname->name, pubname->size,
```

Comment:
Since you're here, can you add error checks to these throughout.

Reply:
Checks to all the Tss2_MU_*_Marshal calls added in https://github.com/tpm2-software/tpm2-tools/pull/2386/commits/6a570acc653b8bdea5063bd398113660b4700dfc